### ISSUE TITLE
pgzip.Reader.Multistream not implemented

### ISSUE THREAD

**Issue by rasky** [klauspost/pgzip#6]
I was looking into using pgzip for [multigz](https://github.com/rasky/multigz) but it looks like the Multistream() method is not implemented.

**Comment by klauspost** [comment]
Ah. Seems like this was added recently. I will look at it at once.

Instead of pgzip, you should maybe look at [compress/gzip](https://github.com/klauspost/compress) which is faster than the standard library gzip for compression (and the inflate has MultiStream). With your method pgzip isn't going to help you, but you can compress multiple parts in parallel.

For decompression, you can get the readahead of pgzip by wrapping the reader in [this library](https://github.com/klauspost/readahead) which I am just in the process of documenting for release.

Have you looked at [bgzf](https://godoc.org/github.com/biogo/hts/bgzf)? It has a slightly different approach, by simply doing flush at regular intervals, so it is separate deflate blocks, but with an offset table at the end, which seems a bit more efficient. 

Also the [pargzip](https://godoc.org/github.com/golang/build/pargzip) output should be compatible with your approach, since it produces separate concatenated gzip files, by default 1MB each.

You should consider larger default block sizes, since the overhead of splitting into 64KB blocks will be quite big. 250KB->1MB will likely be better.

**Comment by klauspost** [comment]
Fixed with commit fccba12d82cf377f4f4fba76e17689af4f676a4f

**Comment by rasky** [comment]
Thanks for the pointer, compress/gzip is probably better for my case, yes. I'll look into it. The readahead module seems also interesting, it's something that I had reimplemented a couple of times as well so props for releasing it :+1: 

I had started with bgzf, but I needed something where the index table was a separate concept compared to the flush intervals. I also needed different ways of deciding where the split the compression stream, so in the end I went ahead and come up with a different library.

I'm inhering the 64kb block default setting from bgzf. I'll look into the performance impact of changing it, I think it mostly depends on the different patterns of usage, and that's why it can be customized by users of the library.
